### PR TITLE
Model danceability

### PR DESCRIPTION
**Pull request by octodoge**
@MadelineKelly196 @15Max
* `model-danceability.ipynb` to train and evaluate danceability model (also works with Colab GPU)
  * Early stopping on validation set to not worry about num epochs
  * Tuning of training hyperparameters (batch size, learning rate) will be made in another branch
  * Used MSE for training (most "natural" choice IMO)

* `utils/models.py` to collect model classes
  * Tuning of the models already there will be made in another branch (also dataset should be recreated larger)
    * The one currently used for danceability is based on a pre-made architecture but doesn't works correctly IMO (it learns correctly but evaluation is noisy, probably a problem with batch normalization, `.eval()` mode and our dataset; same with other pre-made architectures; a simple custom architecture similar to those used for genres seems preferable)

* `utils/model_params/` to collect model parameters (same root dir as `models.py`)
  * Updated `.gitattributes` to track them with Git LFS
  * Currently, `danceability.pth` has Mean Absolute Error of about 0.1 on test set
    * Used MAE for evaluation (most interpretable choice IMO)

@EdoardoCortolezzis
Usage example on web app (`spec` obtained with `utils.preprocessing.transform()`:
```
model = DanceabilityModel()
model.load_state_dict(torch.load(param_path, map_location='cpu'))
model.eval() #disable training mode
danceability = model(spec).item()
```

### CODE REVIEW

**Comment by octodoge** [comment]
Also added logging with Neptune to save info about dataset, model, and training hyperparameters of each run. The idea is to log these things while tuning the hyperparameters so that we can retrieve the configuration that worked best (lowest MAE on test set). IMO we just need to tune:
- model architecture
- batch size
- learning rate
But before starting the tuning we should recreate the dataset with more tracks

**Comment by octodoge** [comment]
Recreated the dataset with 5464 tracks (500+ per genre). I did not get more because git LFS free storage is limited. Also is a pain to upload it to Colab to train with GPU if it's too large. Anyway now should be large enough to get some good results

**Comment by 15Max** [comment]
I agrere with the choice for the loss function since it's the most commonly used for this kind of task.
The final size for the dataset should be good enough, also considering that it's the largest for the limited free space provided

**Review by 15Max — APPROVED** [review]

**Comment by octodoge** [comment]
Update for completeness after tuning of the models: some pre-made architecture actually works as good as a simple custom architecture (eg. squeezenet, alexnet, etc.)
